internal/consensus/state_test.go

```diff
@@ -1596,7 +1596,7 @@ func TestStateLock_POLSafety2(t *testing.T) {
 	ensureNewProposal(proposalCh, height, round)
 
 	ensurePrevote(voteCh, height, round)
-	validatePrevote(t, cs1, round, vss[0], propBlockID1.Hash)
```

Comment:
How that hell this was passing before?

Reply:
Ok, now I know.

The two blocks produced before (`propBlock0` and `propBlock1`) were _identical_. So they matched, meaning that we could prevote the other.

This is a problematic test.

Now, since we don't use the same genesis time for both, instead using the current time, they are identical except for the timestamp. So the correct behavior, prevoting nil, is observed.

Reply:
that is a very problematic test.

Reply:
Added to a pending action on #1731.